packages/block-library/src/navigation-link/edit.js

```diff
@@ -271,6 +277,17 @@ export default function NavigationLinkEdit( {
 		replaceBlock( clientId, newSubmenu );
 	}, [ getBlocks, clientId, selectBlock, replaceBlock, attributes ] );
 
+	// On mount, if this is a new link without a URL and it's selected,
+	// select the parent block (submenu or navigation) instead to keep the appender visible.
+	// This helps us return focus to the appender if the user closes the link ui without creating a link.
+	// If we leave focus on this block, then when we close the link without creating a link, focus will
+	// be lost during the new block selection process.
+	useEffect( () => {
+		if ( isNewLink.current && isSelected && ! url ) {
+			selectBlock( parentBlockClientId );
+		}
+	}, [] );
```

Comment:
We're omitting deps here to have this run "on mount" but I want to check with you that your comfortable that values of `url` and `isSelected` might become stale.

If this is truely a "one time" thing then let's acknowledge that with an eslint comment on the deps line and an explanation. That will help us in future 😅

Reply:
Done!